src/libraries/Microsoft.Extensions.Http/src/LifetimeProcessing/Handlers/Entries/ActiveHandlerTrackingEntry.cs

```diff
@@ -5,71 +5,71 @@
 using System.Diagnostics;
 using System.Threading;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Http.LifetimeProcessing.Handlers.Entries.Base;
 using Microsoft.Extensions.Internal;
+using Timer = System.Timers.Timer;
 
-namespace Microsoft.Extensions.Http
+namespace Microsoft.Extensions.Http.LifetimeProcessing.Handlers.Entries
 {
     // Thread-safety: We treat this class as immutable except for the timer. Creating a new object
     // for the 'expiry' pool simplifies the threading requirements significantly.
-    internal sealed class ActiveHandlerTrackingEntry
+    internal sealed class ActiveHandlerTrackingEntry : HandlerTrackingEntryBase, IDisposable
     {
-        private static readonly TimerCallback _timerCallback = (s) => ((ActiveHandlerTrackingEntry)s!).Timer_Tick();
         private readonly object _lock;
-        private bool _timerInitialized;
+        private bool _isDisposed;
         private Timer? _timer;
-        private TimerCallback? _callback;
+        private Action<ActiveHandlerTrackingEntry>? _callback;
 
         public ActiveHandlerTrackingEntry(
             string name,
             LifetimeTrackingHttpMessageHandler handler,
             IServiceScope? scope,
             TimeSpan lifetime)
+            : base(name, scope)
         {
-            Name = name;
             Handler = handler;
-            Scope = scope;
             Lifetime = lifetime;
 
             _lock = new object();
+            _timer = new Timer(lifetime.TotalMilliseconds)
+            {
+                AutoReset = false
+            };
+            _timer.Elapsed += (_, _) => Timer_Tick();
```

Comment:
This would _unnecessary allocate_ each time an ActiveHandlerTrackingEntry is created.

In general, I am not sure the use of the heavy-weight System.Timers.Timer is justified here.